### DISCUSSION TITLE
Uploading lwt to a webhoster

### DISCUSSION THREAD

**Discussion by horrridd** [HugoFara/lwt#18] [Q&A] [answered]
So, recently I stumbled upon this version of lwt which seems to be much better than original, and I honestly have no idea if it's possible for you to help me solve my problem.
Original lwt supported an option to install it to a webhoster, which I honestly find amazing as I can acess lwt from any place. So I thought I would try to do the same thing with your improved version and see if it works. (I use free 000webhost) Unfortunately there is some problem - when I try to add new text I get an error.

> Fatal Error in SQL Query: LOAD DATA LOCAL INFILE '/storage/ssd2/332/18504332/tmp/tmpti.txt' INTO TABLE temptextitems FIELDS TERMINATED BY '\t' LINES TERMINATED BY '\n' (@w,@c) set TiSeID = @b, TiCount = (@d:=@d+CHAR_LENGTH(@c))+1-CHAR_LENGTH(@c), TiOrder = if(@c like "%\r",case when (@c:=REPLACE(@c,"\r","")) is NULL then NULL when (@b:=@b+1) is NULL then NULL when @d:= @e is NULL then NULL else @a:=@a+1 end, @a:=@a+1), TiText = @c,TiWordCount=@w
> 
> Error Code & Message: [1047] Unsupported 'LOAD DATA LOCAL INFILE' command

Perhaps that's useful, maybe not, but to set it up I followed the instructions provided here: http://dustyfeet.com/info.htm#installhoster

Is it maybe something I can solve by myself?

Thank you in advance

**Comment by HugoFara** [comment]
Hello, thanks for reaching me!

Could you give me more details, for instance complete a [bug report](https://github.com/HugoFara/lwt/issues/new?assignees=&labels=bug&template=bug_report.md&title=LOAD%20DATA%20LOAD%20FILE%20on%20webserver)? It is difficult to solve your problems without the details.

I will do my best to help you though. For now, here are my recommendations:

1. Check the installation procedure ([docs/install.md](https://github.com/HugoFara/lwt/blob/master/docs/install.md)). Particularly, **don't forget to allow read/write** (the ``sudo chmod -R 755 /var/www/html/lwt`` part on Linux).
2. Check your server authorizations. You *might* have a setting saying something "allow local files imports" or "file_uploads".

Let me know of it helped, otherwise we will investigate the issue further :)

**Reply by horrridd** [reply]
Thanks for an answer :)
I'm on Windows, so I don't think the first solution applies, if it does I don't know how to allow it When it comes to second one, as I meantioned all worked fine with the original version of LWT so as far as I'm concerned that also shouldn't be the case. I'll write the bug report as you advised.

**Comment by HugoFara** [comment]
I have been able to reproduce and solve your bug, just edit the php configuration in the ``php.ini`` file. Fin the line
``;mysqli.allow_local_infile = On``
and delete the first semicolon (";") character.

Restart your server and things should be solved.

**Comment by HugoFara** [comment, marked as answer]
Hi again!

I spent some time to writing workarounds to avoid using ``mysqli.allow_local_infile``, because it can lead to vulnerability issues, and some web hosters do not allow changing its value. So, starting from [release 2.5.2](https://github.com/HugoFara/lwt/releases/tag/2.5.2), you do no longer need it! Some parts of the SQL code were rewritten with more PHP, leading to slightly slower database edition but much more security.

I hope it will help you well!
